### DISCUSSION TITLE
Developer Team Management

### DISCUSSION THREAD

**Discussion by YoulSamuel** [community/community#127668] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

Hi, I hope you are all doing great. For this question, please consider the fact that i have no knowledge or background as a developer.  Here is my question
For a company named X that want to hire a developer to work on an existing App/Software but is reluctant to make all the code public to the developer, what are the solution. I do understand that sometimes the developer needs to have access to 100% of the existing code to be able to complete his work, but as more people keeps working on it, one company can't always make his code public. By example, I really doubt that google/Facebook give to their developer full access to their code. 
I made some research and found out about GitHub repository but quickly found out it is just used as a code backup (Please correct me if I'm wrong).
What I'm looking for is a way to: 
1- List and categorized all variables/Api used so that any developer can work on any project without the need to check all the code.
2- Restrict some action based on the developer work (Designer, programmer....)
3- Enable them to work on their portion of code while allowing them to some kind of cloud-based code (only for developer), where they can test it. 
If the code work, I can just copy their code, join it to the mani code and Compile and deploy it.
After much research on google, i have found out that this type of things is called developing environment. Instead of building it from scratch is there a developing platform where one can create an account and start working on its project? 
Or is it recommended to build if from scratch ??? if so, can you provide some advice such the services/platform that might be the best to work with on the long run???
Thanks

**Comment by carlahall678** [comment]
Managing a developer team effectively is crucial for the success of any project. This plan outlines strategies, tools, and best practices for managing the developer team [ raposo infocell ](https://sorecebeinfocell.net/)assigned to Project #127668, ensuring timely delivery, high quality, and team satisfaction.
